blockchain/types/account/externally_owned_account.go

```diff
@@ -67,3 +116,194 @@ func (e *ExternallyOwnedAccount) Equal(a Account) bool {
 
 	return e.AccountCommon.Equal(e2.AccountCommon)
 }
+
+func (e *ExternallyOwnedAccount) GetStorageRoot() common.ExtHash {
+	return e.storageRoot
+}
+
+func (e *ExternallyOwnedAccount) GetCodeHash() []byte {
+	return e.codeHash
+}
+
+func (e *ExternallyOwnedAccount) GetCodeFormat() params.CodeFormat {
+	return e.codeInfo.GetCodeFormat()
+}
+
+func (e *ExternallyOwnedAccount) GetVmVersion() params.VmVersion {
+	return e.codeInfo.GetVmVersion()
+}
+
+func (e *ExternallyOwnedAccount) SetStorageRoot(h common.ExtHash) {
+	e.storageRoot = h
+}
+
+func (e *ExternallyOwnedAccount) SetCodeHash(h []byte) {
+	e.codeHash = h
+}
+
+func (e *ExternallyOwnedAccount) SetCodeInfo(ci params.CodeInfo) {
+	e.codeInfo = ci
+}
+
+func (e *ExternallyOwnedAccount) toSerializable() *externallyOwnedAccountSerializable {
+	return &externallyOwnedAccountSerializable{
+		CommonSerializable: e.AccountCommon.toSerializable(),
+		StorageRoot:        e.storageRoot.Unextend(),
+		CodeHash:           e.codeHash,
+		CodeInfo:           e.codeInfo,
+	}
+}
+
+func (e *ExternallyOwnedAccount) toSerializableExt() *externallyOwnedAccountSerializableExt {
+	return &externallyOwnedAccountSerializableExt{
+		CommonSerializable: e.AccountCommon.toSerializable(),
+		StorageRoot:        e.storageRoot,
+		CodeHash:           e.codeHash,
+		CodeInfo:           e.codeInfo,
+	}
+}
+
+func (e *ExternallyOwnedAccount) fromSerializable(o *externallyOwnedAccountSerializable) {
+	e.AccountCommon.fromSerializable(o.CommonSerializable)
+	e.storageRoot = o.StorageRoot.ExtendZero()
+	e.codeHash = o.CodeHash
+	e.codeInfo = o.CodeInfo
+}
+
+func (e *ExternallyOwnedAccount) fromSerializableExt(o *externallyOwnedAccountSerializableExt) {
+	e.AccountCommon.fromSerializable(o.CommonSerializable)
+	e.storageRoot = o.StorageRoot
+	e.codeHash = o.CodeHash
+	e.codeInfo = o.CodeInfo
+}
+
+func (e *ExternallyOwnedAccount) EncodeRLP(w io.Writer) error {
+	if e.isEOAWithCode() {
+		return rlp.Encode(w, e.toSerializable())
+	}
+	return rlp.Encode(w, e.AccountCommon.toSerializable())
+}
+
+func (e *ExternallyOwnedAccount) EncodeRLPExt(w io.Writer) error {
+	if e.isEOAWithCode() {
+		if e.storageRoot.IsZeroExtended() {
+			return rlp.Encode(w, e.toSerializable()) // EOA without code. [n, b, hR, k, 0x0, eCH, 0x0]
+		}
+		return rlp.Encode(w, e.toSerializableExt()) // EOA with code, has ExtHash [n, b, hR, k, sRExt, cH, cI]
+	}
+	return rlp.Encode(w, e.AccountCommon.toSerializable()) // EOA without code. [n, b, hR, k]
```

Comment:
Please remove EOA with Code and EOA without code from the comments.
```suggestion
			return rlp.Encode(w, e.toSerializable()) // Full encoding. [n, b, hR, k, sR, cH, cI]
		}
		return rlp.Encode(w, e.toSerializableExt()) // Full encoding, has ExtHash [n, b, hR, k, sRExt, cH, cI]
	}
	return rlp.Encode(w, e.AccountCommon.toSerializable()) // Compact encoding. [n, b, hR, k]
```